homeassistant/components/delijn/sensor.py

```diff
@@ -0,0 +1,123 @@
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTRIBUTION = "Data provided by data.delijn.be"
+
+CONF_NEXT_DEPARTURE = 'next_departure'
+CONF_STOP_ID = 'stop_id'
+CONF_API_KEY = 'api_key'
+CONF_NUMBER_OF_DEPARTURES = 'number_of_departures'
+
+DEFAULT_NAME = 'De Lijn'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_API_KEY): cv.string,
+    vol.Required(CONF_NEXT_DEPARTURE): [{
+        vol.Required(CONF_STOP_ID): cv.string,
+        vol.Optional(CONF_NUMBER_OF_DEPARTURES, default=5): cv.positive_int}]
+})
+
+
+async def async_setup_platform(
+        hass, config, async_add_entities, discovery_info=None):
+    """Create the sensor."""
+    api_key = config[CONF_API_KEY]
+    name = DEFAULT_NAME
+
+    session = async_get_clientsession(hass)
+
+    sensors = []
+    for nextpassage in config[CONF_NEXT_DEPARTURE]:
+        stop_id = nextpassage[CONF_STOP_ID]
+        number_of_departures = nextpassage[CONF_NUMBER_OF_DEPARTURES]
+        line = Passages(hass.loop,
+                        stop_id,
+                        number_of_departures,
+                        api_key,
+                        session,
+                        True)
+        await line.get_passages()
+        if line.passages is None:
+            _LOGGER.warning("No data recieved from De Lijn")
+            return
+        sensors.append(DeLijnPublicTransportSensor(line, name))
+
+    async_add_entities(sensors, True)
+
+
+class DeLijnPublicTransportSensor(Entity):
+    """Representation of a Ruter sensor."""
+
+    def __init__(self, line, name):
+        """Initialize the sensor."""
+        self.line = line
+        self._attributes = {}
+        self._name = name
+        self._state = None
+        self._attributes[ATTR_ATTRIBUTION] = ATTRIBUTION
+
+    async def async_update(self):
+        """Get the latest data from the De Lijn API."""
+        await self.line.get_passages()
+        if self.line.passages is None:
+            _LOGGER.warning("No data recieved from De Lijn")
+            return
+        try:
+            attributes = {}
+            first = self.line.passages[0]
+            if first['due_at_realtime'] is not None:
+                first_passage = first['due_at_realtime']
+            else:
+                first_passage = first['due_at_schedule']
+            self._state = first_passage
+
+            self._name = first['stopname']
+            attributes['stopname'] = first['stopname']
+            attributes['line_number_public'] = first['line_number_public']
+            attributes['line_transport_type'] = first['line_transport_type']
+            attributes['final_destination'] = first['final_destination']
+            attributes['due_at_schedule'] = first['due_at_schedule']
+            attributes['due_at_realtime'] = first['due_at_realtime']
+            attributes['due_in_min'] = first['due_in_min']
+            attributes['next_passages'] = self.line.passages
+            self._attributes = attributes
```

Comment:
Update the instance attribute dict with the new attributes instead of replacing it.

Reply:
Just to be sure I understand you correctly:
I need to replace all the attributes['some_key'] with self.attributes['some_key'] instead of creating a dict and then assigning (and thus replacing) it to the self._attributes. Is that understanding correct?

Reply:
OK, I've updated this in the latest commit (as per my description above).